collectors/googlestackdriver/collector.js

```diff
@@ -119,16 +121,24 @@ timestamp < "${state.until}"`;
 
                 // Stackdriver Logging api has some rate limits that we might run into.
                 // If we run inot a rate limit error, instead of returning the error,
-                // we return the state back to the queue with an additional second added, up to 10
+                // we return the state back to the queue with an additional second added, up to 15 min
                 // https://cloud.google.com/logging/quotas
                 // Error: 8 RESOURCE_EXHAUSTED: Received message larger than max (4518352 vs. 4194304),
-                // so half the given interval and if interval is less than 15 sec then try again with same interval.
+                // so half the given interval and if interval is less than 15 sec then reduce the page size to half.
+
                 if(err.code === API_THROTTLING_ERROR){
-                    const nextPollInterval = state.poll_interval_sec < 10 ?
-                        state.poll_interval_sec + 1:
-                        10;
+                    const interval = state.poll_interval_sec < 60 ? 60 : state.poll_interval_sec;
+                    const nextPollInterval = state.poll_interval_sec < MAX_POLL_INTERVAL ?
+                        interval + 60 : MAX_POLL_INTERVAL;
                     const currentInterval = moment(state.until).diff(state.since, 'seconds');
-                    if (currentInterval <= 15) {
+                    if (currentInterval <= 15 && err.details.includes('Received message larger than max')) {
```

Comment:
Is it possible to use some error code instead of sub string search `err.details.includes('Received message larger than max')`? At some point Google can change the error message text.

Reply:
We  already checking error code [here](url). We get two different message for throttling .
1. `{code: 8 ,details : "Quota exceeded for quota metric 'Read requests' and limit 'Read requests per minute' of service 'logging.googleapis.com'" }`-> It resolved by just making the api call after 1 min or so.
2. `{code: 8 ,details : Received message larger than max (4776477 vs. 4194304)} `-> In this scenario we are reducing the page size to handle the error.